Wikipedia/Code/NSURL+WMFLinkParsing.m

```diff
@@ -180,7 +180,13 @@ + (NSURL *)wmf_desktopURLForURL:(NSURL *)url {
 #pragma mark - Properties
 
 - (BOOL)wmf_isWikiResource {
-    return [self.path wmf_isWikiResource];
+    static NSString *wikiResourceSuffix = nil;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        wikiResourceSuffix = [NSString stringWithFormat:@".%@", WMFDefaultSiteDomain];
+    });
+    NSString *lowercaseHost = self.host.lowercaseString;
+    return (!lowercaseHost || [lowercaseHost isEqualToString:WMFDefaultSiteDomain] || [lowercaseHost hasSuffix:wikiResourceSuffix]) && [self.path wmf_isWikiResource];
```

Comment:
Just for reference… our FAQ is on mediawiki.org and follows the same logic. There was a poorly named `wmf_isNonStandardURL` property to check URLs like this that don't have language domains and special case some things.